python/ray/data/_internal/plan.py

```diff
@@ -136,6 +136,8 @@ def __init__(
         # determined by the config at the time it was created.
         self._context = copy.deepcopy(DataContext.get_current())
 
+        self._skip_optimizer_pipeline = not self._context.optimizer_enabled
```

Comment:
Nit, what about naming this variable "_generated_from_pipeline"? This will be easier to reason about. Because it only indicates whether this plan is generated from a pipeline, and doesn't nothing to do with the optimizer here.

Reply:
Good point, updated.